cli/azd/pkg/auth/manager.go

```diff
@@ -432,14 +437,20 @@ func (m *Manager) LoginWithDeviceCode(
 		return nil, err
 	}
 
-	// Display the message to the end user as to what to do next, then block waiting for them to complete
-	// the flow.
-	fmt.Fprintln(deviceCodeWriter, code.Message())
+	console.Message(ctx, fmt.Sprintf("Start by copying the next code: %s", output.WithBold(code.UserCode())))
+	console.Message(ctx, "Then press enter and continue to log in from your browser...")
+	_ = bufio.NewScanner(console.Handles().Stdin).Scan()
 
+	url := "https://login.microsoftonline.com/common/oauth2/deviceauth"
```

Comment:
Do we always want to use the the "common" authority here, or should we be using the tenant-id in some way?

Reply:
Updated to use: https://microsoft.com/devicelogin

When using `tenant-id` like `azd auth login --use-device-code --tenant-id foo` , the tenant-id is used to get the initial user-code, then, when going to `https://microsoft.com/devicelogin`, the user-code will be only to login to the specific tenant.  If we try to use some account which has no access to that tenant, we would get something like:

![image](https://github.com/Azure/azure-dev/assets/24213737/b07c37e1-3044-4c5a-9a68-0c622d0959a2)